src/User/src/Repository/UserRepository.php

```diff
@@ -52,17 +52,6 @@ public function getUsers(array $filters = []): UserCollection
             $qb->andWhere('user.status = :status')->setParameter('status', $filters['status']);
         }
```

Comment:
```suggestion
        } else {
            $qb->andWhere('user.status != :status')->setParameter('status', UserStatusEnum::Deleted);
        }
```
This way, unless filtering by a specific status, we exclude deleted users.